### PR TITLE
Add documentation about our CI pipeline

### PR DESCRIPTION
**Pull request by clinssen**
Following discussion during the NEST Hackathon at 2020-11-11: confusion on my part on what was causing the CI to fail, turned out to be the addition of hardcoded expected number of compiler warnings in #1796. This adds some documentation about NEST-specific parts of CI.

### CODE REVIEW

File: doc/contribute/index.rst [hunk 1]
```diff
@@ -33,6 +33,8 @@ For making changes to the PyNEST APIs, please see our :doc:`templates_styleguide
 If you are a Vim user and require support for SLI files, please refer to our
 :doc:`templates_styleguides/vim_support_sli`.
 
+An explanation of our continuous integration pipeline can be found under :doc:`ci.rst`.
```
Comment (sarakonradi) [inline doc/contribute/index.rst:36]:
```suggestion
An explanation of our continuous integration pipeline can be found under :doc:`ci.`.
```

File: doc/contribute/ci.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#. Static checks
+
+   This first stage checks the code without building or running it.
+
+   - clang-format checks C++ code formatting (whitespace, etc.). Please note that a specific version of clang-format is used for consistency. The maximum line length is set to 120 characters.
+
+   - Vera++ and cppcheck are used for more detailed semantic (but still static) code analysis.
+
+   - pycodestyle is used to statically check Python code. A few errors are intentionally ignored, defined in the variable ``PEP8_IGNORES`` in `extras/static_code_analysis.sh <https://github.com/nest/nest-simulator/blob/master/extras/static_code_analysis.sh>`_. The maximum line length is again 120 characters.
```
Comment (sarakonradi) [inline doc/contribute/ci.rst:31]:
```suggestion
   - pycodestyle is used to statically check Python code. A few errors are intentionally ignored, defined in the variable ``PEP8_IGNORES`` in `extras/static_code_analysis.sh <https://github.com/nest/nest-simulator/blob/master/extras/static_code_analysis.sh>`_. Again, the maximum line length is 120 characters.
```

File: doc/contribute/ci.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Stringent quality control is particularly important in the context of NEST, a neuronal network simulator with the emphasis on correctness, reproducibility and performance. However, given the limited amount of the available resources, it is wasteful to transfer the responsibility to re-run the test suite for all target platforms on every single code base change to the shoulders of the developers.
+
+In order to address this problem, a `Travis <http://travis-ci.org/>`_ Continuous Integration (CI) service has been set up, which allows for regular testing of changes that are getting into the tree and timely reporting of identified problems. This way, issues will be discovered earlier and the amount of efforts to fix them will be significantly decreased (hopefully).
```
Comment (sarakonradi) [inline doc/contribute/ci.rst:8]:
```suggestion
In order to address this problem, a `Travis <http://travis-ci.org/>`_ Continuous Integration (CI) service has been set up, which allows for regular testing of changes that are getting into the tree and timely reporting of identified problems. This way, issues will be discovered earlier and the amount of efforts to fix them significantly decreased.
```

**Review by sarakonradi — CHANGES_REQUESTED** [review]
Thanks for adding the documentation! Here are some minor comments.

**Comment by clinssen** [comment]
Cheers, no comments from me, all three suggestions merged!

**Review by terhorstd — APPROVED** [review]
LGTM :+1: 
Thanks, @clinssen!

File: doc/contribute/index.rst [hunk 4]
```diff
@@ -33,6 +33,8 @@ For making changes to the PyNEST APIs, please see our :doc:`templates_styleguide
 If you are a Vim user and require support for SLI files, please refer to our
 :doc:`templates_styleguides/vim_support_sli`.
 
+An explanation of our continuous integration pipeline can be found under :doc:`ci.`.
```
Comment (sarakonradi) [inline doc/contribute/index.rst:36]:
```suggestion
An explanation of our continuous integration pipeline can be found under :doc:`ci`.
```

Comment (sarakonradi) [reply doc/contribute/index.rst:36]: Spotted an extra full stop. Now the link should work!

**Review by sarakonradi — CHANGES_REQUESTED** [review]

**Comment by clinssen** [comment]
Oops, cheerio!

**Review by sarakonradi — APPROVED** [review]
Awesome, thanks!

**Pull request merged**